test/topology/test_change_ip.py

```diff
@@ -11,47 +11,69 @@
 import pytest
 from test.pylib.util import wait_for_cql_and_get_hosts, wait_for
 from test.pylib.random_tables import Column, IntType, TextType
+from test.pylib.internal_types import IPAddress
+from cassandra.pool import Host # type: ignore # pylint: disable=no-name-in-module
 import time
 logger = logging.getLogger(__name__)
 
 
 @pytest.mark.asyncio
-async def test_change_two(manager, random_tables):
+async def test_change_two(manager, random_tables, mode):
     """Stop two nodes, change their IPs and start, check the cluster is
     functional"""
     servers = await manager.running_servers()
+    cql = manager.get_cql()
+    s_0 = servers[0].server_id
+    s_1 = servers[1].server_id
+    s_2 = servers[2].server_id
+    s2_old_ip = servers[2].ip_addr
+
     table = await random_tables.add_table(name='t1', pks=1, columns=[
         Column("pk", IntType),
         Column('int_c', IntType)
     ])
-    s_1 = servers[1].server_id
-    s_2 = servers[2].server_id
-    logger.info("Gracefully stopping servers %s and %s to change ips", s_1, s_2)
+
+    logger.info(f"Servers {servers}, gracefully stopping servers {s_1} and {s_2} to change ips")
     await manager.server_stop_gracefully(s_1)
     await manager.server_stop_gracefully(s_2)
-    await manager.server_change_ip(s_1)
-    await manager.server_change_ip(s_2)
+    s1_new_ip = await manager.server_change_ip(s_1)
+    s2_new_ip = await manager.server_change_ip(s_2)
+    logger.info(f"s1 new ip {s1_new_ip}, s2 new ip {s2_new_ip}")
+
+    async def wait_for_peers(host: Host, expected_peers: set[IPAddress]):
+        logger.info(f"waiting for {host} to see its peers {expected_peers}")
+        async def see_proper_peers():
+            current_peers = set(r.peer for r in await cql.run_async("select peer from system.peers", host=host))
+            if current_peers == expected_peers:
+                return True
+            logger.info(f"waiting for {host} to see its peers, current {current_peers}, expected {expected_peers}")
+        await wait_for(see_proper_peers, time.time() + 60)
+
+    # We're checking the crash scenario here - the servers[0] crashes just after
+    # saving s1_new_ip but before removing s1_old_ip. After its restart we should
+    # see s1_new_ip.
+    if mode != 'release':
+        await manager.api.enable_injection(servers[0].ip_addr, 'crash-before-prev-ip-removed', one_shot=True)
     await manager.server_start(s_1)
+    if mode != 'release':
+        s0_logs = await manager.server_open_log(s_0)
+        await s0_logs.wait_for('crash-before-prev-ip-removed hit, killing the node')
+        await manager.server_stop(s_0)
+        await manager.server_start(s_0)
+    host0 = (await wait_for_cql_and_get_hosts(cql, [servers[0]], time.time() + 60))[0]
+    await wait_for_peers(host0, {s1_new_ip, s2_old_ip})
```

Comment:
Previously (the inlined version of) `wait_for_peers` was called after reconnecting driver, which served as a workaround for https://github.com/scylladb/python-driver/issues/295 among other things.

Now you're calling it after a server restarts + wait_for_cql_and_get_hosts --- so the test might now likely hit https://github.com/scylladb/python-driver/issues/295, making it flaky.

I'll merge, the flakiness should be rare, but please send a follow-up to address this (we need a driver reconnect after server restart before doing queries)

Reply:
done